kaspi_app/SwiftUI/ProductSwiftUI.swift

```diff
@@ -0,0 +1,184 @@
+        Product(image: Image("iron"),name: "Pelican Strem че...", price: 50000, category: "Утюги", bonus: "5%", credit0012: false, action: doNothing),
+        Product(image: Image("iron"),name: "Pelican Strem че...", price: 50000, category: "Утюги", bonus: "5%", credit0012: true, action: doNothing),
+        Product(image: Image("iron"),name: "Pelican Strem че...", price: 50000, category: "Утюги", bonus: "5%", credit0012: true, action: doNothing),
+        Product(image: Image("iron"),name: "Pelican Strem че...", price: 50000, category: "Утюги", bonus: "5%", credit0012: false, action: doNothing),
+       ]
+    
+    var body: some View{
+        VStack(alignment: .leading){
+            
+            Text("Вы недавно смотрели")
+                .font(Font.system(size: 18,weight: .bold))
+                .padding(.horizontal)
+            
+            ScrollView(.horizontal,showsIndicators: false){
+                HStack(spacing: 18){
+                        ForEach(lastVisited) { visit in
+                            ProductView(product: visit,type:.small)
+                            }
+                }.padding(.horizontal)
+            }
+            Text("Все предложения")
+            .font(Font.system(size: 18,weight: .bold))
+            .foregroundColor(Color(red: 0.2, green: 0.61, blue: 0.9))
+            .padding(.horizontal)
+        }
+    }
+}
+
+struct RecView: View{
+    var lastVisited = [
+        Product(image: Image("iron"),name: "Pelican Strem че...", price: 50000, category: "Утюги", bonus: "5%", credit0012: false, action: doNothing),
+        Product(image: Image("iron"),name: "Pelican Strem че...", price: 50000, category: "Утюги", bonus: "5%", credit0012: false, action: doNothing),
+        Product(image: Image("iron"),name: "Pelican Strem че...", price: 50000, category: "Утюги", bonus: "5%", credit0012: true, action: doNothing),
+        Product(image: Image("iron"),name: "Pelican Strem че...", price: 50000, category: "Утюги", bonus: "5%", credit0012: true, action: doNothing),
+        Product(image: Image("iron"),name: "Pelican Strem че...", price: 50000, category: "Утюги", bonus: "5%", credit0012: false, action: doNothing),
+       ]
+    
+    var body: some View{
+        VStack(alignment: .leading){
+            
+            Text("Вас могут заинтересовать")
+                .font(Font.system(size: 18,weight: .bold))
+            HStack{
+                VStack{
+                    ProductView(product: lastVisited[0],type:.large)
+                    ProductView(product: lastVisited[3],type:.large)
+                }
+                
+                VStack{
+                    ProductView(product: lastVisited[3],type:.large)
+                    ProductView(product: lastVisited[1],type:.large)
+                    
+                }
+            }
+        }
+    }
+}
+
+
+struct ProductView: View {
+    let product:Product
+    let type: ProductCardType
+    var heigt: CGFloat{
+        switch type {
```

Comment:
it is okay. But also you can use this shorter version instead of switch statement, because you have only one case:
return (typy == .large) ? 233 : 195

But it will not work properly if you will have more than 2 cases